### ISSUE TITLE
Unusable parameter log for EMDTransport

### ISSUE THREAD

**Issue by tardyb** [rflamary/POT#43]
Hi,

I need to get some values from transport computation, as cost matrix, value of minimisation...
Some of these values are stored in the log. But when I do:
```
ot_emd,log = ot.da.EMDTransport(norm="max",log=0)
```
I get the following error:
```
TypeError: __init__() got an unexpected keyword argument 'log'
```
In the EMDTransport class declaration there is:
```
"""
Parameters
----------
...
log : int, optional (default=0)
Controls the logs of the optimization algorithm
..."""
 ```
So the question:
Is it voluntary not to be able to recover the log with this class  ? And so to get it back I should directly call the emd function without using the EMDTransport class.

Another question:
I want to get the min value computed by the minimisation problem (first with EMD but also with sinkhorn) to find a link between effectiveness of transport and OA obtained in classification, how can I do and are there some others values usable to get this kind of information?

At the end my goal is estimate several transports and choose automatically the best.

Regards

Benjamin

**Comment by rflamary** [comment]
Hello Benjamin,

You are right the return of the log is not implemented at the moment and the documentation is false. It is not voluntary and it can be added similarly to SinkhornL1l2Transport that has a working implementation.

If you want to do it we welcome Pull requests, or else we will definitely address this bug in the future but no promises on quick implementation. A quick way to get the log is indeed to call directly the emd function.

Conserning your other question, the min value of the OT problem is probably not a good metric to see if the OA accuracy is good. It will be minimum for EMD and increase with any regularization. The way we did choose the best transport in our papers is through validation (which is difficult in DA). One good validation is called circular validation where you do the transport from source to target and then from target to source with labels obtained through target classifier and measure how well the classes are preserved when you come back to source.

Rémi

**Comment by tardyb** [comment]
Thank you for your quick response.
I have modified the two classes (EMD and LpL1) where log is not used.
I've just finished and I will open a PR soon. 

Concerning the other question I didn't explain it well, but I would retain your remark for future work
For the moment, I only consider one transport method after another.
An example of what I doing: 
I have six different ( from different years) source datasets, and only one target domain, in context of land cover classification problem.
In this case I will consider the six EMDTransport and compare them to choose the best, in this case I would that several information from transport  be usable. If not I will do circular validation, but It could be a heavy process in addition to a current heavy process.

Thank you for your time.
